torch/do_regr.py

```diff
@@ -0,0 +1,188 @@
+        self.output_size = output_size
+        self.fc1 = nn.Linear(input_size, 115)  # linear: y = Ax + b
+        self.fc2 = nn.Linear(115, 78)
+        self.fc3 = nn.Linear(78, 26)
+        self.fc4 = nn.Linear(26, 46)
+        self.fc5 = nn.Linear(46, 82)
+        self.fc6 = nn.Linear(82, 106)
+        self.fc7 = nn.Linear(106, output_size)
+        
+    def forward(self, x):
+        x = self.fc1(x)
+        x = F.relu(x)
+        x = self.fc2(x)
+        x = F.relu(x)
+        x = self.fc3(x)
+        x = F.relu(x)
+        x = self.fc4(x)
+        x = F.relu(x)
+        x = self.fc5(x)
+        x = F.relu(x)
+        x = self.fc6(x)
+        x = F.relu(x)
+        x = self.fc7(x)
+        
+        output = x = F.relu(x)
+        return output
+
+
+def scale_features(df: pd.DataFrame, model_dir: Path):
+    """Scale table's features for regression.
+
+    Args:
+        df (pd.DataFrame): Table of features to scale.
+        model_dir (Path): Model directory to get the scaler files.
+
+    Returns:
+        torch.FloatTensor: Tensor of features that can be directly used for
+        model evaluation.
+    """
+    def scale_features_core(values, n):
+        # column_values = df[column].values.reshape(-1, 1)
+        values = np.array([values]).reshape(1, -1)
+        scaler_file = model_dir/'scalers'/f'xscaler_{n:02}.pkl'
+        with open(scaler_file, 'rb') as sf:
+            xscaler = pickle.load(sf)
+            xscaler.clip = False
+            return xscaler.transform(values).item()
+
+    columns = [df[column].apply(scale_features_core, args=(n,)) for n, column in enumerate(df.columns, start=1)]
+    scaled_df = pd.concat(columns, axis=1)
+
+    # columns = [scale_features_core(n, column) for n, column in enumerate(data_table.columns, start=1)]
+    # scaled_df = pd.DataFrame(np.hstack(columns), columns=data_table.columns)
+
+    return torch.FloatTensor(scaled_df.to_numpy())
+
+
+def scale_targets(data_table: pd.DataFrame, label_names: list, scale_exp=[1.0, 14.0, 14.0, 16.0, 0.0]):
+    """Scale target data (from simulation results).
+
+    Target data is in original scaling, whereas the training predictions are
+    scaled down by 10^n. Exponents used to scale the data down for 
+    training (n) are stored as a pickle file in the model folder. 
+    The pickle file is a list of powers, which are used to reverse the scaling.
+
+    Args:
+        data_table (pd.DataFrame): DataFrame of target data.
+        label_names (list): List of column names to scale.
+        scale_exp (list, optional): List of parameter exponents. 
+            Defaults to [1.0, 14.0, 14.0, 16.0, 0.0].
+
+    Returns:
+        pd.DataFrame: DataFrame of scaled target data.
+    """
+
+    # label_names = ['potential (V)', 'Ne (#/m^-3)', 'Ar+ (#/m^-3)', 'Nm (#/m^-3)', 'Te (eV)']
+    for i, column in enumerate(label_names):
+        data_table.update(data_table.iloc[:, i]/(10**scale_exp[i]))
+
+    return data_table
```

Comment:
Shouldn't this be scaled_df??
```suggestion
    return scaled_df
```